src/runtime/Memory.cpp

```diff
@@ -99,11 +99,13 @@ class ReverseArrayIterator {
     T* _ptr;
 };
 
-void Memory::init(ExecutionState& state, DataSegment* source, uint32_t dstStart, uint32_t srcStart, uint32_t srcSize)
+bool Memory::init(ExecutionState& state, DataSegment* source, uint32_t dstStart, uint32_t srcStart, uint32_t srcSize)
 {
-    checkAccess(state, dstStart, 0, srcSize);
+    if (checkAccess(state, dstStart, 0, srcSize)) {
+        return true;
```

Comment:
It is strange for me if a function returns true when failed. I would prefer the opposite.